alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt

```diff
@@ -525,12 +547,321 @@ class SecureMonitorRestApiIT : AlertingRestTestCase() {
+        )
+
+        createRandomEmailAccountWithGivenName(true, randomAlphaOfLength(10))
+
+        try {
+            val request = """
+                {
+                  "from": 0,
+                  "size": 20,
+                  "sort": { "email_group.name.keyword": "desc" },
+                  "query": {
+                    "bool": {
+                      "must": {
+                        "match_all": {}
+                      }
+                    }
+                  }
+                }
+            """.trimIndent()
+            val searchEmailAccountResponse = userClient?.makeRequest(
+                "POST",
+                "${AlertingPlugin.EMAIL_ACCOUNT_BASE_URI}/_search",
+                StringEntity(request, ContentType.APPLICATION_JSON),
+                BasicHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+            )
+            assertEquals("Search Email failed", RestStatus.OK, searchEmailAccountResponse?.restStatus())
+        } finally {
+            deleteRoleAndRoleMapping(TEST_HR_ROLE, ALERTING_SEARCH_EMAIL_ACCOUNT_ACCESS)
+        }
+    }
+
+    fun `test search email accounts with an user without search email account role`() {
+
+        createUserWithTestDataAndCustomRole(
+            user,
+            TEST_HR_INDEX,
+            TEST_HR_ROLE,
+            TEST_HR_BACKEND_ROLE,
+            getClusterPermissionsFromCustomRole(ALERTING_NO_ACCESS_ROLE)
+        )
+
+        createRandomEmailAccountWithGivenName(true, randomAlphaOfLength(5))
+
+        try {
+            val request = """
+                {
+                  "from": 0,
+                  "size": 20,
+                  "sort": { "email_group.name.keyword": "desc" },
+                  "query": {
+                    "bool": {
+                      "must": {
+                        "match_all": {}
+                      }
+                    }
+                  }
+                }
+            """.trimIndent()
+            userClient?.makeRequest(
+                "POST",
+                "${AlertingPlugin.EMAIL_ACCOUNT_BASE_URI}/_search",
+                StringEntity(request, ContentType.APPLICATION_JSON),
+                BasicHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+            )
+        } catch (e: ResponseException) {
+            assertEquals("Unexpected status", RestStatus.FORBIDDEN, e.response.restStatus())
+        } finally {
+            deleteRoleAndRoleMapping(TEST_HR_ROLE, ALERTING_NO_ACCESS_ROLE)
+        }
+    }
+
+    fun `test index email accounts with an user with index email account role`() {
+        createUserWithTestDataAndCustomRole(
+            user,
+            TEST_HR_INDEX,
+            TEST_HR_ROLE,
+            TEST_HR_BACKEND_ROLE,
+            getClusterPermissionsFromCustomRole(ALERTING_INDEX_EMAIL_ACCOUNT_ACCESS)
+        )
+
+        try {
+            val emailAccount = createRandomEmailAccountWithGivenName(true, randomAlphaOfLength(5))
```

Comment:
There is no call to the API. This is not testing if the email account is being indexed.